content/WebAuthn/WebAuthn_Developer_Guide/best-practices.adoc

```diff
@@ -0,0 +1,76 @@
+
+The more security keys an individual has registered per account, the less likely that all of them are lost. However, if this does happen, the identity proofing process should be performed again as rigorously as the first time. Remote proofing technologies and processes can accelerate the process, although in-person identity proofing may be required for sensitive scenarios.
+
+
+=== Mobile / Portable Root of Trust ===
+
+The credentials on a mobile device-based platform authenticator cannot be transferred to another device when the mobile device is replaced. By using a security key as a portable root of trust, users can  bootstrap a new mobile device and ensure there is no disruption in access. WebAuthn can thus be used to remove SMS-based authentication from mobile applications.
+
+
+=== Conditional Authentication ===
+
+In high-risk situations, a security key can be used to quickly validate that the user is in possession of the physical security key and is intentionally performing the action. Prompting the user to touch the security key before proceeding enables validation that the request has come from the user instead of from a compromised system.
+
+
+=== Support ===
+
+WebAuthn support is built into Android-based phones and can be leveraged by developers building mobile applications on that platform. At this time (09-2019), applications on iPhones require the integration of Yubico’s iOS mobile SDK to take advantage of WebAuthn. Note that the RP can query the user’s browser to see if it supports WebAuthn.
+
+
+== Register a Credential ==
+
+Registering a security key is a simple process, but end-users who are new to the process will need clear instructions. Many successful sites have developed animated graphics or little videos that show a user manipulating a security key in the expected fashion (for example, inserting it into a USB port or touching a cell phone with it) and touching the device when required. These animations are very helpful in guiding new users through the process and securing adoption of the new practices.
+
+
+== Manage Credentials ==
+
+In addition to attestation certificates, security keys contain metadata with the following information at minimum:
+
+* The name of the key manufacturer
+* The device type
+* Supported transports
+* Image URL.
+
+=== RP ===
+
+It is important for the RP to capture this information as it enables device validation, user experience customization, and authenticator metrics reports. Identification information needs to include:
+
+* Name of the authenticator
+* Time and location of last use
+
+The name given by the end-user to the security key is stored by the RP and is unique within the RP's system (for any given RP, there can be only one key called "Roger", for example).
```

Comment:
No, my understanding is that within the RP's system, there can be only key called "Roger" or "Fiona" - if someone else has already registered "Fiona" then my key has to be "Fiona01" or whatever.

Reply:
Oh, sorry, looks like I didn't post this comment...

Ok. That's not how it usually works, though, at least assuming we're talking about nicknames chosen by the user. It is of course within the RP's power to require authenticator nicknames to be unique, but there's little reason to do so since their purpose is to help the user differentiate within their own set of authenticators.

Credential IDs, on the other hand, should indeed be globally unique, but they're created by the authenticator hardware and not intended for human consumption.

Reply:
It was nicknames chosen by the user. But now that I bring it up, even I recognize that it doesn't make sense, so I will remove it.